test/battle-test.mocha.js

```diff
@@ -0,0 +1,98 @@
+const fs = require('fs');
+const assert = require('assert');
+
+window = global;
+
+// Without making these modules, the best we can do is directly include them into this workspace.
+eval('' + fs.readFileSync(`${__dirname}/../js/battle-scene-stub.js`));
+eval('' + fs.readFileSync(`${__dirname}/../js/battle-dex.js`));
+eval('' + fs.readFileSync(`${__dirname}/../js/battle-dex-data.js`));
+eval('' + fs.readFileSync(`${__dirname}/../js/battle.js`));
+
+describe('Battle', function () {
+
+	it('should instantiate without issue', function () {
+		var battle = new Battle();
+	});
+
+	it('should process a bunch of messages properly', function () {
+		var battle = new Battle();
+		battle.debug = true;
+
+		[
+			"|init|battle",
+			"|title|FOO vs. BAR",
+			"|j|FOO",
+			"|j|BAR",
+			"|request|",
+			"|player|p1|FOO|169",
+			"|player|p2|BAR|265",
+			"|teamsize|p1|6",
+			"|teamsize|p2|6",
+			"|gametype|singles",
+			"|gen|7",
+			"|tier|[Gen 7] Random Battle",
+			"|rated|",
+			"|seed|",
+			"|rule|Sleep Clause Mod: Limit one foe put to sleep",
+			"|rule|HP Percentage Mod: HP is shown in percentages",
+			"|",
+			"|start",
+			"|switch|p1a: Leafeon|Leafeon, L83, F|100/100",
+			"|switch|p2a: Gliscor|Gliscor, L77, F|242/242",
+			"|turn|1",
+		].forEach(msg => battle.add(msg));
```

Comment:
You're looking for `battle.setQueue`.

Reply:
Ah, I see.

What would you recommend doing for the second block (where the results depend on the events in the first block)?

Reply:
We still don't have a better API than `fastForwardTo` at the moment.